lib/core/src/Cardano/Pool/Ranking.hs

```diff
@@ -104,11 +106,46 @@ desirability
     -> Double
 desirability constants pool
     | f_saturated <= c = 0
-    | otherwise    = (f_saturated - c) * (1 - m)
+    | otherwise        = (f_saturated - c) * (1 - m)
   where
     f_saturated = saturatedPoolRewards constants pool
     m = getRatio $ margin pool
-    c = fromIntegral $ getLovelace $ cost pool
+    c = fromIntegral $ getQuantity $ cost pool
+
+-- | The saturation-level of a pool indicate how far a pool is from the
+-- desirated relative stake fixed by the network. A saturation level above 1
+-- means that the pool is satured. A level of 0.5 / 50% means that the pool owns
+-- half the ideal stake.
+--
+-- The saturation corresponds therefore to the ratio between the ideal relative
+-- stake of a pool on the actual relative stake of that pool.
+--
+-- The ideal relative stake is given by:
+--
+--              1     S     1
+--   σ_ideal = --- * --- = --- = z0 (where S stands for the total available
+--              S     k     k        stake in Ada)
+--
+-- which gives us the saturation as:
+--
+--             σ         σ
+--    sat = --------- = ----
+--           σ_ideal     z0
+--
+saturation
+    :: EpochConstants
+    -> Quantity "lovelace" Word64
+        -- ^ Total delegated stake
+    -> Quantity "lovelace" Word64
+        -- ^ Pool's stake
+    -> Double
+saturation constants total own =
+    σ / z0
```

Comment:
Looks right. 

Wondering slightly if `Ranking` is really the best module for this. Maybe it is.